docs/preview/yamlgettingstarted-publish.md

```diff
@@ -0,0 +1,42 @@
+# YAML getting started - Publish artifact
+
+## Publish artifact step
+
+The `publishArtifact` step can be used to upload a directory, as an artifact associated with the build.
+
+## Example: Publish the artifact staging directory
+
+In the following example, the first step creates a file under the artifact staging directory.
+
+The `publishArtifact` step uploads the artifact staging directory. When the build completes, the artifact named `drop` can be downloaded from the artfacts tab on the build summary page.
+
+```yaml
+steps:
+- script: echo hello > $(build.artifactStagingDirectory)/hello.txt
+
+- publishArtifact: drop
+```
+
+## Example: Publish a custom folder
+
+In the following example, a custom folder is used to stage the artifact.
+
+```yaml
+steps:
+- script: |
+    mkdir layoutDir
+    echo hello > layoutDir/hello.txt
+    echo world > layoutDir/world.txt
+
+- publishArtifact: layout
+  sourceFolder: layoutDir # or $(system.defaultWorkingDirectory)/layoutDir
```

Comment:
how about path instead of sourceFolder

Reply:
oops I originally meant to put `sourcePath`.

I don't mind changing this to simply `path`. My thinking was to make way for `targetPath` for things like file share artifacts, or even the current server artifacts can use a user-specified path to put it under a specific folder. Multiple publish steps can be used today to contribute to a single artifact, if the subpaths are different.